### PR TITLE
CWeapon done, fixes

### PR DESCRIPTION
**Pull request by Fire-Head**

### CODE REVIEW

File: src/peds/PlayerPed.h [hunk 1]
```diff
@@ -20,8 +20,8 @@ class CPlayerPed : public CPed
 	uint8 m_nEvadeAmount;
 	int8 field_1367;
 	uint32 m_nSpeedTimer;
-	int32 m_nShotDelay;
-	float field_1376;	// m_fAttackButtonCounter?
+	uint32 m_nHitAnimDelayTimer;
+	float m_fAttackButtonCounter;	// m_fAttackButtonCounter?
```
Comment (ShFil119) [inline src/peds/PlayerPed.h:24]: comment could be removed

File: src/weapons/Weapon.cpp [hunk 2]
```diff
@@ -30,24 +98,2032 @@ CWeapon::Initialise(eWeaponType type, int ammo)
...
+					
+					CParticle::AddParticle(PARTICLE_BULLETHIT_SMOKE, smokePos, CVector(0.0f, 0.0f, 0.0f));
+					
+					break;
+				}
+				case ENTITY_TYPE_VEHICLE:
+				{
+					((CVehicle *)victim)->InflictDamage(shooter, m_eWeaponType, info->m_nDamage);
+					
+					for ( int32 i = 0; i < 16; i++ )
```
Comment (ShFil119) [inline src/weapons/Weapon.cpp:1015]: +1

File: src/render/Glass.cpp [hunk 3]
```diff
@@ -404,6 +404,7 @@ CGlass::AskForObjectToBeRenderedInGlass(CEntity *entity)
 void
 CGlass::RenderEntityInGlass(CEntity *entity)
 {
+	ASSERT(entity!=NULL);
```
Comment (erorcun) [inline src/render/Glass.cpp:407]: using nil instead of NULLs would be cool

File: src/weapons/Weapon.cpp [hunk 4]
```diff
@@ -30,24 +98,2032 @@ CWeapon::Initialise(eWeaponType type, int ammo)
...
+			}
+			else
+			{				
+				switch ( victim->m_type )
+				{
+					case ENTITY_TYPE_VEHICLE:
+					{
+						((CVehicle *)victim)->InflictDamage(shooter, m_eWeaponType, info->m_nDamage);
+					
+						for ( int32 i = 0; i < 16; i++ )
```
Comment (ShFil119) [inline src/weapons/Weapon.cpp:1264]: +1

File: src/weapons/Weapon.cpp [hunk 5]
```diff
@@ -30,24 +98,2032 @@ CWeapon::Initialise(eWeaponType type, int ammo)
...
+						
+						CParticle::AddParticle(PARTICLE_BULLETHIT_SMOKE, smokePos, CVector(0.0f, 0.0f, 0.0f));
+						
+						break;
+					}
+					
+					case ENTITY_TYPE_BUILDING:
+					case ENTITY_TYPE_OBJECT:
+					{
+						for ( int32 i = 0; i < 16; i++ )
```
Comment (ShFil119) [inline src/weapons/Weapon.cpp:1279]: +1

File: src/weapons/Weapon.cpp [hunk 6]
```diff
@@ -30,24 +98,2032 @@ CWeapon::Initialise(eWeaponType type, int ammo)
...
+	
+	CVector2D front(cam->Front.x, cam->Front.y);
+	front.Normalise();
+	
+	DoBulletImpact(shooter, victim, &source, &target, &point, front);
+
+	CVector bulletPos;
+	if ( CHeli::TestBulletCollision(&source, &target, &bulletPos, 4) )
+	{
+		for ( int32 i = 0; i < 16; i++ )
```
Comment (ShFil119) [inline src/weapons/Weapon.cpp:1582]: +1

**Comment by aap** [comment]
We don't have to put everything in config.h and make everything a define. some hardcoded numbers that aren't likely to change are ok.

**Pull request merged**